lib/ansible/cli/galaxy.py

```diff
@@ -969,39 +991,111 @@ def _display_role(gr):
+                roles_path = context.CLIARGS['roles_path']
+                for path in roles_path:
+                    role_path = os.path.expanduser(path)
+                    if not os.path.exists(role_path):
+                        warnings.append("- the configured path %s does not exist." % role_path)
+                        continue
+                    elif not os.path.isdir(role_path):
+                        warnings.append("- the configured path %s, exists, but it is not a directory." % role_path)
+                        continue
+                    display.display('# %s' % role_path)
+                    path_files = os.listdir(role_path)
+                    path_found = True
+                    for path_file in path_files:
+                        gr = GalaxyRole(self.galaxy, self.api, path_file, path=path)
+                        if gr.metadata:
+                            _display_role(gr)
+
+        elif context.CLIARGS['type'] == 'collection':
+            collections_search_paths = context.CLIARGS['collections_path']
+            collection_name = context.CLIARGS['collection']
+
+            # TODO: Should we support listing all collections in a given namespace?
+
+            # list a specific collection
+            if collection_name:
+                for path in collections_search_paths:
+                    collection_path = GalaxyCLI._resolve_path(path)
+                    if not os.path.exists(path):
+                        if path in C.COLLECTIONS_PATHS:
+                            # don't warn for missing default dirs
+                            continue
+                        warnings.append("- the configured path {0} does not exist.".format(collection_path))
+                        continue
+                    elif not os.path.isdir(collection_path):
+                        warnings.append("- the configured path {0}, exists, but it is not a directory.".format(collection_path))
+                        continue
+
+                    path_found = True
+                    display.display("# {0}".format(path))
+
+                    # Make sure we look inside the 'ansible_collections' dir
+                    if os.path.split(collection_path)[1] != 'ansible_collections':
+                        collection_path = os.path.join(collection_path, 'ansible_collections')
+
+                    try:
+                        namespace, collection = collection_name.split('.')
+                    except ValueError:
+                        raise AnsibleOptionsError("Incorrect value for collection name. Must be 'namespace.collection' got '%s'" % collection_name)
+
+                    b_collection_path = to_bytes(os.path.join(collection_path, namespace, collection), errors='surrogate_or_strict')
+
+                    if not os.path.exists(b_collection_path):
+                        warnings.append("- unable to find {0} in collection paths".format(collection_name))
```

Comment:
Ooo, good catch. I didn't have multiple collection paths in my test environment. I added that and now I see what you're talking about.

I think it should only warn if it's not found in _any_ of the paths and not warn for paths where it didn't exist.

Honestly, it may also make sense to warn if the same collection exists in _multiple_ paths, but I'm not going there for now.

Reply:
Fixed.